### DISCUSSION TITLE
Sometimes Workspace isn't showing in Explorer panel

### DISCUSSION THREAD

**Discussion by drmrbrewer** [coder/code-server#6492] [General]
With the latest code-server v4.17.1 I'm finding that sometimes the Workspace folders don't show up in the Explorer panel to the left (and the clock icon is stuck):

![image](https://github.com/coder/code-server/assets/9009792/b3337e46-a000-424a-8547-b3615e9136a4)

When I close the page and re-open it (possibly more than once), they do eventually show up:

![image](https://github.com/coder/code-server/assets/9009792/f914c3d5-1462-4ed7-9dd6-685d9c6bb188)

I don't recall this ever happening before upgrading to 4.17.1 :-/

**Comment by code-asher** [comment]
Might be this issue https://github.com/coder/code-server/issues/3061#issuecomment-1760022312

**Reply by drmrbrewer** [reply]
Not sure, but it happens a lot.  I just had to open up code-server six times before I got one which actually showed all my workspace folders correctly :-/

**Reply by drmrbrewer** [reply]
@code-asher this issue appears to be gone in `v4.19.0` (at least I haven't seen it yet).

**Reply by code-asher** [reply]
That is great news!  Fingers crossed it has been resolved for the others as well.  :crossed_fingers:  Please let me know if it crops back up.

**Reply by drmrbrewer** [reply]
@code-asher I spoke too soon :-(  It's still happening.  I can't spot any pattern to when it happens or doesn't, and the console log output looks the same for when it loads fine and when it doesn't.  Ah well, maybe something will by chance fix this in a future version.

**Reply by code-asher** [reply]
Thank you for the update!  I think we have it fixed now (see release candidate), although the fix involves removing a patch that causes a breaking change with user settings.  https://github.com/coder/code-server/issues/3061#issuecomment-1823540072

**Reply by drmrbrewer** [reply]
@code-asher I look forward to trying the fix.  I'm actually finding that sometimes, even when the workspace explorer eventually loads, I will have lost my terminal setup, back to just a single terminal... hopefully that is related and will be caught up in this fix too.

**Reply by code-asher** [reply]
Oh interesting that sounds plausibly related.  I am trying to revert the fix I made since it is a breaking change so I will keep an eye out for the terminal issue as well as I test a different fix.

**Reply by drmrbrewer** [reply]
@code-asher so far so good with [v4.19.1](https://github.com/coder/code-server/releases/tag/v4.19.1) :-)

**Reply by code-asher** [reply]
Fantastic news!!
